### PR TITLE
Enable empty startTime, disable past startTime

### PR DESCRIPTION
**Pull request by NateWilliams2**

### CODE REVIEW

**Comment by p4u** [comment]
This PR introduces the possibility to set startBlock:=0 in order to start the process on the next block where the newProcess transaction is included. Maybe it could modify some behavior of this PR?

**Comment by NateWilliams2** [comment]
> Maybe it could modify some behavior of this PR?

The current behavior is
- if startDate is empty, set startBlock to 0 and start the process asap
- if startDate is not empty, ensure it is in the future. If not, return an error

@p4u Would you propose startBlock to be set to 0 in the second scenario, rather than returning an error?

File: urlapi/entity.go [hunk 1]
```diff
@@ -459,6 +464,10 @@ func (u *URLAPI) createProcessHandler(msg *bearerstdapi.BearerStandardAPIdata,
 		return fmt.Errorf("could not retrieve integrator from db: %w", err)
 	}
 
+	currentBlockHeight, _, _ := u.vocClient.GetBlockTimes()
+	if startBlock > 1 && startBlock < currentBlockHeight+10 {
```
Comment (jordipainan) [inline urlapi/entity.go:468]: Why 10 is used ? Can we add it as const ?

Comment (NateWilliams2) [reply urlapi/entity.go:468]: yes, actually what should be used here? Maybe I misremembered 10 blocks as being the standard acceptable waiting time

**Comment by p4u** [comment]
> 

Yes

**Review by p4u — APPROVED** [review]

File: urlapi/entity.go (hunk 1 above)
Comment (jordipainan) [reply urlapi/entity.go:468]: Ahh ok, is used for the possible delay :+1: 
So yeah I think 10 is fine, but add as const as I think will be easier to find and change in the future is required

Comment (p4u) [reply urlapi/entity.go:468]: There is not such limit in the Vochain logic. But a secufity margin is recomended. I think something like 5 blocks would be enough

**Pull request merged**